tools/serverpod_cli/lib/src/util/model_helper.dart

```diff
@@ -22,46 +30,93 @@ class ModelHelper {
-          Directory(joinAll(config.protocolSourcePathParts));
-      protocolSourceFileList =
-          await protocolSourceDir.list(recursive: true).toList();
-    } on PathNotFoundException catch (_) {}
+    var modelSources = <ModelSource>[];
 
-    List<FileSystemEntity> modelSourceFileList = [];
-    try {
-      var modelSourceDir = Directory(joinAll(config.modelSourcePathParts));
-      modelSourceFileList = await modelSourceDir.list(recursive: true).toList();
-    } on PathNotFoundException catch (_) {}
+    var modelSource = await _loadYamlModelsFromDisk(
+        defaultModuleAlias, _absolutePathParts(config.modelSourcePathParts));
+    modelSources.addAll(modelSource);
 
-    var sourceFileList = [...protocolSourceFileList, ...modelSourceFileList];
+    modelSource = await _loadYamlModelsFromDisk(
+        defaultModuleAlias, _absolutePathParts(config.protocolSourcePathParts));
+    modelSources.addAll(modelSource);
+
+    for (var module in config.modulesDependent) {
+      modelSource = await _loadYamlModelsFromDisk(
+          module.nickname, module.modelSourcePathParts);
+      modelSources.addAll(modelSource);
+
+      modelSource = await _loadYamlModelsFromDisk(
+          module.nickname, module.protocolSourcePathParts);
+      modelSources.addAll(modelSource);
+    }
 
     // TODO This sort is needed to make sure all generated methods
     // are in the same order. Move this logic to the code generator instead.
-    sourceFileList.sort((a, b) => a.path.compareTo(b.path));
+    modelSources
+        .sort((a, b) => a.yamlSourceUri.path.compareTo(b.yamlSourceUri.path));
 
-    var files = sourceFileList.whereType<File>().where(
-        (file) => modelFileExtensions.any((ext) => file.path.endsWith(ext)));
+    return modelSources;
+  }
+
+  static List<String> _absolutePathParts(List<String> pathParts) {
+    return split(absolute(joinAll(pathParts)));
+  }
+
+  static Future<List<ModelSource>> _loadYamlModelsFromDisk(
+    String moduleAlias,
+    List<String> pathParts,
+  ) async {
+    var files = await _loadAllModelFiles(pathParts);
 
     List<ModelSource> sources = [];
     for (var model in files) {
       var yaml = await model.readAsString();
 
       sources.add(ModelSource(
+        moduleAlias,
         yaml,
         model.uri,
-        extractPathFromModelRoot(config, model.uri),
+        extractPathFromModelRoot(pathParts, model.uri),
       ));
     }
 
     return sources;
   }
 
+  static Future<Iterable<File>> _loadAllModelFiles(
+    List<String> absolutePathParts,
+  ) async {
+    List<FileSystemEntity> modelSourceFileList = [];
+
+    var path = joinAll(absolutePathParts);
+
+    if (!isAbsolute(path)) {
+      path =
+          Platform.isWindows ? '$separator$separator$path' : '$separator$path';
+    }
+
+    try {
+      var modelSourceDir = Directory(path);
+      modelSourceFileList = await modelSourceDir.list(recursive: true).toList();
+    } on PathNotFoundException catch (_) {}
```

Comment:
Question: Is there a risk that this PathNotFoundException might be propagated from one of the subdirectories instaed of from the directory we are inspecting? Thinking about broken symlinks or the likes?

Maybe it would be better just to explicitly inspect of the directory exists before listing it and only skip in that case?